script/20240807-ir-recover/20240807-ir-recover.s.sol

```diff
@@ -226,23 +230,74 @@ contract Migration__20240807_IR_Recover is Migration {
   }
 
+  function _fake_pause() internal {
+    address pauseEnforcer = 0xe514d9DEB7966c8BE0ca922de8a064264eA6bcd4;
+    console.log("Pranking Pause Enforcer");
+    vm.prank(pauseEnforcer);
+    (bool success,) = address(_mainchainGW).call(abi.encodeWithSignature("pause()"));
+    require(success, "Cannot pause mainchain gateway");
+    console.log("Stop pranking Pause Enforcer");
+  }
+
   function _perform_checkAfterPrankFix() internal {
+    console.log("=== _perform_checkAfterPrankFix ===");
     // - Total weight in `BM` and `GW` the same
     {
       uint256 totalWeightBM = _mainchainBM.getTotalWeight();
       uint96 totalWeightGW = getGWTotalWeight();
       require(totalWeightBM == uint256(totalWeightGW), "Mismatched total weight");
+      require(totalWeightBM == 2200, "Mismatched total weight 2200");
     }
 
     // - Weight of all operators in `BM` and `GW` the same
     (, address[] memory operatorsBM, uint96[] memory weightsBM) = _mainchainBM.getFullBridgeOperatorInfos();
     for (uint256 i = 0; i < operatorsBM.length; i++) {
       require(getGWWeight(operatorsBM[i]) == weightsBM[i], "Mismatched weight");
+      require(getGWWeight(operatorsBM[i]) == 100, "Mismatched weight 100");
     }
 
     {
       _postCheck_Withdrawable();
     }
+
+    // Check minimum weight = specific number
+    {
+      require(IQuorum(address(_mainchainGW)).minimumVoteWeight() == 1540, "Mismatched minimum vote weight 1540");
+    }
+
+    // Check threshold
+    {
+      require(IQuorum(address(_mainchainGW)).checkThreshold(1540) == true, "Malfunction threshold 1540");
+      require(IQuorum(address(_mainchainGW)).checkThreshold(1541) == true, "Malfunction threshold 1541");
+      require(IQuorum(address(_mainchainGW)).checkThreshold(1539) == false, "Malfunction threshold 1539");
+      require(IQuorum(address(_mainchainGW)).checkThreshold(0) == false, "Malfunction threshold 0");
+
+      require(IWithdrawalLimitation(address(_mainchainGW)).checkHighTierVoteWeightThreshold(1980) == true, "Malfunction high tier threshold 1980");
+      require(IWithdrawalLimitation(address(_mainchainGW)).checkHighTierVoteWeightThreshold(1981) == true, "Malfunction high tier threshold 1981");
+      require(IWithdrawalLimitation(address(_mainchainGW)).checkHighTierVoteWeightThreshold(1979) == false, "Malfunction high tier threshold 1979");
+      require(IWithdrawalLimitation(address(_mainchainGW)).checkHighTierVoteWeightThreshold(0) == false, "Malfunction high tier threshold 0");
+    }
+
+    // Check `depositForBatch` is removed
+    {
+      _fake_unpause();
+      Transfer.Request[] memory dummyRequests = new Transfer.Request[](1);
+
+      // Method get removed, so it reverted in fallback with invalid deposit
+      vm.expectRevert(abi.encodeWithSignature("ErrInvalidInfo()"));
+      address(_mainchainGW).call(abi.encodeWithSignature("requestDepositForBatch((address,address,(uint8,uint256,uint256))[])", dummyRequests));
```

Comment:
This test case will always return false and is not suitable to check anything. Please provide a new post-test or remove this. Note: Using the fallback will cause all valid batch deposits to succeed. One way to test this scenario is to check if the gateway should only emit one event for batch requests greater than 1 when calling `requestDepositForBatch`.

Reply:
updated